dissect/target/plugins/apps/texteditor/windowsnotepad.py

```diff
@@ -0,0 +1,182 @@
+import zlib
+from typing import Iterator, List, Union
+
+from dissect.cstruct import cstruct
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import (
+    UnixUserRecord,
+    WindowsUserRecord,
+    create_extended_descriptor,
+)
+from dissect.target.plugin import export
+from dissect.target.plugins.apps.texteditor.texteditor import (
+    GENERIC_TAB_CONTENTS_RECORD_FIELDS,
+    TexteditorPlugin,
+)
+
+c_def = """
+struct multi_block_entry {
+    uint16    offset;
+    uleb128   len;
+    wchar     data[len];
+    char      crc32[4];
+};
+
+struct single_block_entry {
+    uint16    offset;
+    uleb128   len;
+    wchar     data[len];
+    char      unk1;
+    char      crc32[4];
+};
+
+struct header_crc {
+    char      unk[4];
+    char      crc32[4];
+};
+
+struct tab {
+    char                        magic[3];         // NP\x00
+    char                        header_start[2];  // \x00\x01
+    uleb128                     fsize1;
+    uleb128                     fsize2;
+    char                        header_end[2];    // \x01\x00
+};
+"""
+TextEditorTabRecord = create_extended_descriptor([UserRecordDescriptorExtension])(
+    "texteditor/windowsnotepad/tab", GENERIC_TAB_CONTENTS_RECORD_FIELDS
+)
+
+c_windowstab = cstruct()
+c_windowstab.load(c_def)
+
+
+def _calc_crc32(data: bytes) -> bytes:
+    """Perform a CRC32 checksum on the data and return it as bytes."""
+    return zlib.crc32(data).to_bytes(length=4, byteorder="big")
```

Comment:
I assume the reason you return this as bytes is because for some reason the checksums are big endian? Maybe a comment in the structure would be nice if that's the case.